compiler/src/dotty/tools/dotc/ast/MainProxies.scala

```diff
@@ -94,8 +94,20 @@ object MainProxies {
       val body = Try(call, handler :: Nil, EmptyTree)
       val mainArg = ValDef(nme.args, TypeTree(defn.ArrayType.appliedTo(defn.StringType)), EmptyTree)
         .withFlags(Param)
+      /** Replace typed `Ident`s that have been typed with a TypeSplice with the reference to the symbol.
+       *  The annotations will be retype-checked in another scope that may not have the same imports.
+       */
+      def insertTypeSplices = new TreeMap {
+          override def transform(tree: Tree)(using Context): Tree = tree match
+            case tree: Ident if tree.symbol.exists => ref(tree.symbol)
```

Comment:
If this works this seems simpler:
```suggestion
            case tree: Ident => TypedSplice(tree)
```

Reply:
This failed with

```scala
Error:  102 |            case tree: Ident => TypedSplice(tree)
Error:      |                                            ^^^^
Error:      |                   Found:    (tree : dotty.tools.dotc.ast.untpd.Ident)
Error:      |                   Required: dotty.tools.dotc.ast.tpd.Tree
```

Reply:
ah, you probably need to do:
If this works this seems simpler:
```suggestion
            case tree: tpd.Ident => TypedSplice(tree)
```